### PR TITLE
Fix MSVC MINMAX CI Failures

### PR DESCRIPTION
**Pull request by mborland**
Many of these errors continue to stem from https://github.com/boostorg/random/pull/76 not being resolved.

### CODE REVIEW

**Comment by mborland** [comment]
@jzmaddock This gets us back to green.

File: test/test_carlson.hpp [hunk 1]
```diff
@@ -4,6 +4,15 @@
...
 
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#ifdef min
+#undef min
+#endif
+#ifdef max
+#undef max
+#endif
```
Comment (jzmaddock) [inline test/test_carlson.hpp:15]: Point of pedantry: isn't this superfluous now?

Comment (mborland) [reply test/test_carlson.hpp:15]: You are correct.

**Comment by mborland** [comment]
@jzmaddock This is good with the superfluous macros removed. I will open a PR with random to hit all the minmax violations.

**Comment by jzmaddock** [comment]
Thanks Matt!

**Pull request merged**